### PR TITLE
Add AWS Kendra Client tool

### PR DESCRIPTION
**Pull request by collindutter**

### CODE REVIEW

File: griptape/labs/tools/aws_kendra/tool.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+@define
+class AwsKendraClient(BaseTool):
+    index_id: str = field(default=None, kw_only=True)
+
+    @activity(config={
+        "name": "query",
+        "description": "Can be used to query an AWS Kendra index.",
+        "schema": Schema({
+            Literal(
+                "queryText",
```
Comment (vasinov) [inline griptape/labs/tools/aws_kendra/tool.py]: Could you rename it to `query_text`?

File: griptape/labs/tools/aws_kendra/tool.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+from griptape.artifacts import BaseArtifact, TextArtifact, ErrorArtifact
+from griptape.core import BaseTool
+from griptape.core.decorators import activity
+from schema import Schema, Literal
+from attr import define, field
+import boto3
+
+@define
+class AwsKendraClient(BaseTool):
+    index_id: str = field(default=None, kw_only=True)
```
Comment (vasinov) [inline griptape/labs/tools/aws_kendra/tool.py]: Will it work with `None`? If so the type should be `Optional[str]`. If not, then no default?

File: griptape/labs/tools/aws_kendra/tool.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            ): str,
+        })
+    })
+    def query(self, params: dict) -> BaseArtifact:
+        try:
+            kendra = boto3.client('kendra')
+            query_text = params['values']["queryText"]
+            response = kendra.query(IndexId=self.index_id, QueryText=query_text)
+            response_text = ''
+            for query_result in response["ResultItems"]:
```
Comment (vasinov) [inline griptape/labs/tools/aws_kendra/tool.py:36]: Should we split the result into a `ListArtifact` of `TextArtifact`s? If not, then we should probably separate documents with `\n\n`, so it's easier to chunk later.

**Review by vasinov — CHANGES_REQUESTED** [review]
How hard would it be to add a test with a mock?

File: griptape/labs/tools/aws_kendra/tool.py (hunk 2 above)
Comment (collindutter) [reply griptape/labs/tools/aws_kendra/tool.py]: `index_is` is required, updating.

File: griptape/labs/tools/aws_kendra/tool.py (hunk 3 above)
Comment (collindutter) [reply griptape/labs/tools/aws_kendra/tool.py:36]: Didn't know we had a ListArtifact, done!

**Comment by collindutter** [comment]
@vasinov 
> How hard would it be to add a test with a mock?

Moto doesn't support Kendra. Would a test similar to https://github.com/griptape-ai/griptape-labs/blob/main/tests/unit/test_aws_pricing.py be sufficient?

**Comment by vasinov** [comment]
> @vasinov
> 
> > How hard would it be to add a test with a mock?
> 
> Moto doesn't support Kendra. Would a test similar to https://github.com/griptape-ai/griptape-labs/blob/main/tests/unit/test_aws_pricing.py be sufficient?

Yeah, I think that works. Could also use a proper mock like [HERE](https://github.com/griptape-ai/griptape/blob/577ccc7064437b6bccc80d3a47246df35f3e81a5/tests/integration/drivers/storage/vector/test_pinecone_vector_storage_driver.py#L13).

File: griptape/labs/tools/aws_kendra/tool.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import boto3
+
+
+@define
+class AwsKendraClient(BaseTool):
+    index_id: str = field(kw_only=True)
+
+    @activity(
+        config={
+            "name": "query",
```
Comment (vasinov) [inline griptape/labs/tools/aws_kendra/tool.py:17]: Names in config are no longer required.

**Pull request closed without merging**